cnd/src/btsieve/ethereum.rs

```diff
@@ -50,7 +50,32 @@ where
         matching_transaction_and_receipt(blockchain_connector, start_of_swap, |transaction| {
             // transaction.to address is None if, and only if, the transaction
             // creates a contract.
-            transaction.to.is_none() && transaction.input == bytecode
+
+            let is_contract_creation = transaction.to.is_none();
+            let is_expected_contract = &transaction.input == bytecode;
+
+            if !is_contract_creation {
+                tracing::trace!("rejected because transaction doesn't create a contract");
+            }
+
+            if !is_expected_contract {
+                tracing::trace!("rejected because contract code doesn't match");
+
+                // only compute levenshtein distance if we are on trace level, converting to hex is expensive at this scale
+                if tracing::level_enabled!(tracing::level_filters::LevelFilter::TRACE) {
+                    let actual = hex::encode(&transaction.input);
+                    let expected = hex::encode(&bytecode);
+
+                    let distance = levenshtein::levenshtein(&actual, &expected);
+
+                    // We probably need to find a meaningful value here, expiry is 4 bytes.
```

Comment:
Can we remove this comment and create an issue instead?

Reply:
Ticket created already yesterday, will remove comment.